File: caption_generator.py
import random

def generate_caption():
    cat_captions = [
        "Pure joy", "Cat life", "Stay pawsitive", "Meow moment", "Stay fuzzy",
        "Fur baby", "Purrfect view", "Stay cozy", "Sweet kitty", "Cat nap",
        "Little lion", "Mood always", "Soft paws", "Whiskers Wednesday", "Peak cozy",
        "Totally obsessed", "Simply cute", "Daily dose", "Meow magic", "Kitty love",
        "Tail wags", "Furry friend", "Staying pawsome", "Golden hour", "Best bud",
        "Wild child", "Heart melter", "Cuddle bug", "Lazy days", "Too cute",
        "Cat vibe", "Always sleepy", "Purr machine", "Little explorer", "Dream big",
        "Simply purrfect", "Main character", "Fluff ball", "Tiny hunter", "Vibe check",
        "Maximum fluff", "So soft", "Mischief maker", "Current mood", "Sweet dreams",
        "Paws up", "Just chilling", "Cat logic", "Home buddy", "Stay curious",
        "Happy kitty", "Feline fine", "Eye contact", "Double trouble", "Pure fluff",
        "Living large", "Small fry", "Cloud nine", "Snack time", "Playful soul",
        "Big energy", "Spirit animal", "Cuteness overload", "Iconic energy", "Zero thoughts",
        "Zoomies time", "Floof alert", "Staring contest", "Sun bather", "Nap queen",
        "Nap king", "Stay wild", "Morning ritual", "Graceful hunter", "Gentle giant",
        "Tiny beans", "Toe beans", "Sassy soul", "Truly blessed", "My muse",
        "Cat energy", "Perfect timing", "Loaf mode", "Standard procedure", "Simply iconic",
        "Furry chaos", "Weekend vibes", "Quiet moments", "Little shadow", "Always watching",
        "Pure bliss", "Life partner", "Best view", "Total sweetheart", "Completely relaxed",
        "High alert", "Natural beauty", "Living fancy", "Pet therapy", "Unconditional love"
    ]
    random_selection = random.sample(cat_captions, 5)
    return random_selection

File: server/services/captionGenerator.js
const DEFAULT_CAPTIONS = [
    'Meow', 'Purrfect', 'Cozy', 'Vibes', 'Sweet', 'Fluffy', 'Floof',
    'Paws', 'Chilling', 'Mood', 'Cute', 'Silly', 'Adorable', 'Sleepy',
    'Playful', 'Wow', 'Love', 'Kitty', 'Cat', 'Feline', 'Whiskers',
    'Beans', 'Snack', 'Hunter', 'Staring', 'Zoomies', 'Chaos', 'Blessed',
    'Muse', 'Energy', 'Iconic', 'Fancy', 'Therapy', 'Joy', 'Magic',
    'Soft', 'Wild', 'Heart', 'Bug', 'Lazy', 'Vibe', 'Machine', 'Explorer',
    'Dream', 'Hunter', 'Check', 'Soft', 'Maker', 'Current', 'Dreams',
    'Logic', 'Buddy', 'Curious', 'Happy', 'Fine', 'Contact', 'Trouble',
    'Large', 'Fry', 'Nine', 'Time', 'Soul', 'Energy', 'Animal', 'Overload',
    'Energy', 'Thoughts', 'Alert', 'Contest', 'Bather', 'Queen', 'King',
    'Wild', 'Ritual', 'Giant', 'Beans', 'Soul', 'Timing', 'Mode',
    // Funny, Meme, & Fighting related
    'Bruh', 'Sus', 'Derp', 'Yeet', 'Sheesh', 'NPC', 'Oof', 'LOL',
    'Bonk', 'Smack', 'Bap', 'Pow', 'Ninja', 'KO', 'Wasted', 'Boss',
    'Fatality', 'Oops', 'Glitch', 'Sigma', 'Pounce', 'Sneak', 'Help',
    'Epic fail', 'RKO', 'Based', 'W cat', 'L cat', 'Skill issue', 'Got em',
    'Cursed', 'He attac', 'He protec', 'Oh lawd', 'Chonk', 'Angy', 'Busted',
    'Paw slap', 'Combo', 'Player 2', 'Game over', 'Try hard', 'Caught lacking',
    'Side eye', 'Bombastic', 'Bro chill', 'Scrap', 'Takedown', 'Victory',
    'Ratio', 'Sneak attack', 'RIP', 'Gamer', 'No way', 'Why tho', 'Flawless'
];

function generateCaptions(count = 5) {
    const source = [...DEFAULT_CAPTIONS];
    const picks = [];

    while (picks.length < count && source.length > 0) {
        const idx = Math.floor(Math.random() * source.length);
        picks.push(source.splice(idx, 1)[0]);
    }

    while (picks.length < count) {
        // Fallback random captions (mostly 1 word, max 2) if we run out of unique DFAULT_CAPTIONS
        const fallbackWords = ['Meow', 'Cute', 'Funny', 'Sweet', 'Fluffy', 'Vibin', 'Paws', 'Cozy', 'Chill', 'Purr', 'Love', 'Wow', 'Silly', 'Soft', 'Chilling', 'Mood', 'Adorable', 'Sleepy', 'Playful', 'Bruh', 'Bonk', 'Yeet', 'Oof', 'KO', 'RKO', 'Based', 'Chonk', 'Epic fail', 'Paw slap', 'Combo'];
        picks.push(fallbackWords[Math.floor(Math.random() * fallbackWords.length)]);
    }

    return picks;
}

function getRandomCaption() {
    const source = [...DEFAULT_CAPTIONS];
    const fallbackWords = ['Meow', 'Cute', 'Funny', 'Sweet', 'Fluffy', 'Vibin', 'Paws', 'Cozy', 'Chill', 'Purr', 'Love', 'Wow', 'Silly', 'Soft', 'Chilling', 'Mood', 'Adorable', 'Sleepy', 'Playful', 'Bruh', 'Bonk', 'Yeet', 'Oof', 'KO', 'RKO', 'Based', 'Chonk', 'Epic fail', 'Paw slap', 'Combo'];
    
    // 50/50 chance to pick from DEFAULT_CAPTIONS or fallbackWords
    if (Math.random() > 0.5) {
        return source[Math.floor(Math.random() * source.length)];
    }
    return fallbackWords[Math.floor(Math.random() * fallbackWords.length)];
}

module.exports = { generateCaptions, getRandomCaption };
